apps/src/lib/tools/jsdebugger/DebugConsole.jsx

```diff
@@ -212,8 +212,10 @@ export default connect(
     displayOutputToConsole() {
       if (this.props.logOutput.size > 0) {
         return this.props.logOutput.map((output, i) => {
-          if (typeof output === 'object') {
+          try {
             output = output.toJS();
```

Comment:
I'm missing some context here: Where does the `toJS()` function come from?  Is the exception we're catching `TypeError: output.toJS is not a function` as [mentioned here](https://github.com/code-dot-org/code-dot-org/pull/28140/files#r282277643), or is it the `TypeError: Converting circular structure to JSON` shown in this PR's description?

Reply:
The `output.toJS is not a function` is referring to the circular structure in `converting circular structure to JSON`. So the exception that we're catching is `TypeError: output.toJS is not a function` in this case. Those outputs are either instances of sprite, camera, or group which are all circular structures.

Reply:
I meant to ask about the happy-path case: When this code does work, where does the `toJS()` function come from? The only library I'm familiar with that provides this function is [Immutable](https://immutable-js.github.io/immutable-js/) and, to my knowledge, we're not using it with the debug console.

If the underlying issue here can be identified by the presence or absence of the `toJS` function, it might be nice to write your solution this way, to minimize duplication and not depend on error handling for control flow:

```js
if ('function' === typeof output.toJS) {
  output = output.toJS();
}
return <Inspector key={i} data={output} />;
```

If, however, `toJS` consistently exists but sometimes fails due to circular structure, try/catch is probably our best bet here.  In that case my only recommendation would be to leave the catch statement a no-op, so lines 218 and 220 aren't duplicated.

Reply:
`toJS` seems to be only absent for circular structures as far as I know so checking the presence of `toJS` function makes sense! Thanks for letting me know. I've made the changes and pushed it to this PR.